tests/unit/customizations/s3/test_fileinfo.py

```diff
@@ -95,6 +120,15 @@ def test_no_raise_md5_with_kms(self):
         # The file should have been saved.
         self.assertTrue(os.path.isfile(self.filename))
 
+    def test_no_raise_md5_when_md5_unavailable(self):
+        self.response_data['ETag'] = '"0"'
+        self.response_data['ServerSideEncryption'] = 'AES256'
```

Comment:
It will cause `_can_validate_md5_with_etag` to pass, so that the only thing preventing MD5 is the `MD5_AVAILABLE` flag. That way if the test fails, it is because that flag did not get set or checked properly.